crates/commonware-node/src/dkg/manager/actor/mod.rs

```diff
@@ -160,1035 +137,1023 @@ where
-        let metadata: Metadata<ContextCell<TContext>, U64, Bytes> = Metadata::init(
-            context.with_label("database"),
-            metadata::Config {
-                partition: format!("{}_database", config.partition_prefix),
-                codec_config: RangeCfg::from(0..=usize::MAX),
-            },
-        )
-        .await
-        .expect("must be able to initialize metadata on disk to function");
-
-        let db = MetadataDatabase::new(metadata);
-
-        {
-            // TODO: set this when opening the database/metadata for the first time?
-            let mut tx = DkgReadWriteTransaction::new(db.read_write());
-            tx.set_node_version(env!("CARGO_PKG_VERSION").to_string());
-            tx.commit()
-                .await
-                .wrap_err("failed to commit init transaction")?;
-        }
-
-        let syncing_players = Gauge::default();
-
-        let peers = Gauge::default();
-
-        context.register(
-            "syncing_players",
-            "how many syncing players were registered; these will become players in the next ceremony",
-            syncing_players.clone(),
-        );
-
-        context.register(
-            "peers",
-            "how many peers are registered overall for the latest epoch",
-            peers.clone(),
-        );
-
-        let ceremony = ceremony::Metrics::register(&context);
-
-        let metrics = Metrics {
-            peers,
-            syncing_players,
-            ceremony,
-        };
+        let metrics = Metrics::init(&context);
 
         Ok(Self {
             config,
             context,
             mailbox,
-            db,
             metrics,
-            finalized_tip: None,
-            gaps: vec![],
-            pending_gap: None.into(),
-            pending_finalized_block: None,
-
-            pending_dkg_outcome_requests: HashMap::new(),
-            notarized_fetch_to_requests: HashMap::new(),
-            notarized_fetch_abort_handles: HashMap::new(),
-            notarized_stream: FuturesUnordered::new(),
         })
     }
 
+    pub(crate) fn start(
+        mut self,
+        dkg_channel: (
+            impl Sender<PublicKey = PublicKey>,
+            impl Receiver<PublicKey = PublicKey>,
+        ),
+    ) -> Handle<()> {
+        spawn_cell!(self.context, self.run(dkg_channel).await)
+    }
+
     async fn run(
         mut self,
         (sender, receiver): (
             impl Sender<PublicKey = PublicKey>,
             impl Receiver<PublicKey = PublicKey>,
         ),
     ) {
-        let mut tx = DkgReadWriteTransaction::new(self.db.read_write());
-
-        // Emits an error event on return.
-        if self.initialize_epoch_state(&mut tx).await.is_err() {
+        let Ok(initial_state) = read_initial_state_from_genesis(
+            &mut self.context,
+            &self.config.execution_node,
+            self.config.epoch_length,
+            self.config.initial_share.clone(),
+        )
+        .await
```

Comment:
this won't work with a full node since we can't provide this historical state. we either read directly from genesis or we fetch the state from the EL last finalized block